### ISSUE TITLE
[BUG]: cat: can't open 'GEN23-RTPCR-0829-NTC-3-1b/amended_consensus/*.fa': No such file or directory

### ISSUE THREAD

**Issue by Codes1985** [CFIA-NCFAD/nf-flu#47]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description of the Bug/Issue

Hello,

I have come across an edge-case related to the sequencing of negative controls samples.

On occasion, the read set for an NTC sample exceeds the minimum number of reads (default: 100), and are allowed to be processed by the pipeline.  The issue is related to IRMA and how it handles samples with low numbers of reads.

The pipeline will crash when IRMA has an output as follows:

Loading config file 'irma_config.sh'
[2023-09-11 14:32:12]	IRMA/FLU-minion started run 'GEN23-RTPCR-0829-NTC-3-1b'
[2023-09-11 14:32:12]	IRMA/FLU-minion found 1306.5T free space, only needed ~6.0M
[2023-09-11 14:32:12]	IRMA/FLU-minion pre-processed
[2023-09-11 14:32:12]	IRMA/FLU-minion R1 started (253)
[2023-09-11 14:32:13]	IRMA/FLU-minion R1 all-match with BLAT finished
[2023-09-11 14:32:13]	IRMA/FLU-minion R1 consolidated & cleaned
[2023-09-11 14:32:13]	IRMA/FLU-minion R1 aborted, no matches found
[2023-09-11 14:32:13]	IRMA/FLU-minion converted back to fastq
[2023-09-11 14:32:13]	IRMA/FLU-minion saved unmatched read patterns
[2023-09-11 14:32:13]	IRMA/FLU-minion skipping final assembly, no reference files found
[2023-09-11 14:32:15]	IRMA/FLU-minion moving project
[2023-09-11 14:32:15]	IRMA/FLU-minion finished!

Essentially there are no influenza reads, so it doesn't generate any outputs, which appears to affect lines 38-42 in the `irma.nf` file:

```
IRMA $irma_module $reads $meta.id

  if [ -d "${meta.id}/amended_consensus/" ]; then
    cat ${meta.id}/amended_consensus/*.fa > ${meta.id}.irma.consensus.fasta
  fi
```
Resulting in the following error that crashes the pipeline:

`cat: can't open 'GEN23-RTPCR-0829-NTC-3-1b/amended_consensus/*.fa': No such file or directory`

However, when IRMA encounters a sample with, I'm assuming, with at least one RP or read the outputs generared are enough to allowe the pipeline to proceed:

Loading config file 'irma_config.sh'
[2023-09-11 14:32:18]	IRMA/FLU-minion started run 'GEN23-RTPCR-0829-NTC-3-1a'
[2023-09-11 14:32:18]	IRMA/FLU-minion found 1306.5T free space, only needed ~5.7M
[2023-09-11 14:32:18]	IRMA/FLU-minion pre-processed
[2023-09-11 14:32:18]	IRMA/FLU-minion R1 started (164)
[2023-09-11 14:32:18]	IRMA/FLU-minion R1 all-match with BLAT finished
[2023-09-11 14:32:19]	IRMA/FLU-minion R1 consolidated & cleaned
[2023-09-11 14:32:19]	IRMA/FLU-minion R1 sorted using BLAT
[2023-09-11 14:32:19]	IRMA/FLU-minion R1 aborted, found fewer than 3 RPs or 3 reads for all templates
[2023-09-11 14:32:19]	IRMA/FLU-minion moving project
[2023-09-11 14:32:19]	IRMA/FLU-minion finished!

For now, to mitigate this problem, I added `errorStrategy = 'ignore'` to the `modules_nanopore.config`:

```
withName: 'IRMA' {
    // increased job time limit for IRMA process to accommodate large samples
    time = '24h'
    errorStrategy = 'ignore'
    publishDir = [
      [
        path: { "${params.outdir}/irma"},
        saveAs: { filename -> filename.equals('versions.yml') ? null : filename },
        mode: params.publish_dir_mode
      ],
      [
        path: { "${params.outdir}/consensus/irma/" },
        pattern: "*.irma.consensus.fasta",
        mode: params.publish_dir_mode
      ]
    ]
  }
```
However, I'm not sure if this is the most robust way around this particular issue.

Thank you very much for your support.

### Nextflow command-line

```shell
nextflow nf-flu_v3.3.4/cfia-ncfad-nf-flu-3.3.4/workflow/main.nf --input IRVC20230831IHN-1.csv --platform nanopore --outdir IRVC20230831IHN-1_nf-flu_results --major_allele_fraction 0.25 -profile singularity,slurm -resume
```

### Error Message

```shell
ERROR ~ Error executing process > 'NF_FLU:NANOPORE:IRMA (GEN23-RTPCR-0829-NTC-3-1b)'

Caused by:
  Process `NF_FLU:NANOPORE:IRMA (GEN23-RTPCR-0829-NTC-3-1b)` terminated with an error exit status (1)

Command executed:

  touch irma_config.sh
  echo 'SINGLE_LOCAL_PROC=16' >> irma_config.sh
  echo 'DOUBLE_LOCAL_PROC=8' >> irma_config.sh
  # default tmp in current working directory instead of defaulting to /tmp 
  # which may be restricted in size on HPC clusters
  echo 'ALLOW_TMP=1' >> irma_config.sh
  echo 'TMP=$PWD' >> irma_config.sh
  if [ true ]; then
    echo 'DEL_TYPE="NNN"' >> irma_config.sh
    echo 'ALIGN_PROG="BLAT"' >> irma_config.sh
  fi
  
  IRMA FLU-minion GEN23-RTPCR-0829-NTC-3-1b.merged.fastq.gz GEN23-RTPCR-0829-NTC-3-1b
  
  if [ -d "GEN23-RTPCR-0829-NTC-3-1b/amended_consensus/" ]; then
    cat GEN23-RTPCR-0829-NTC-3-1b/amended_consensus/*.fa > GEN23-RTPCR-0829-NTC-3-1b.irma.consensus.fasta
  fi
  ln -s .command.log GEN23-RTPCR-0829-NTC-3-1b.irma.log
  cat <<-END_VERSIONS > versions.yml
  "NF_FLU:NANOPORE:IRMA":
     IRMA: $(IRMA | head -n1 | sed -E 's/^Iter.*IRMA\), v(\S+) .*/\1/')
  END_VERSIONS

Command exit status:
  1

Command output:
  Loading config file 'irma_config.sh'
  [2023-09-11 16:06:21]	IRMA/FLU-minion started run 'GEN23-RTPCR-0829-NTC-3-1b'
  [2023-09-11 16:06:21]	IRMA/FLU-minion found 1305.9T free space, only needed ~6.0M
  [2023-09-11 16:06:21]	IRMA/FLU-minion pre-processed
  [2023-09-11 16:06:21]	IRMA/FLU-minion R1 started (253)
  [2023-09-11 16:06:22]	IRMA/FLU-minion R1 all-match with BLAT finished
  [2023-09-11 16:06:23]	IRMA/FLU-minion R1 consolidated & cleaned
  [2023-09-11 16:06:23]	IRMA/FLU-minion R1 aborted, no matches found
  [2023-09-11 16:06:23]	IRMA/FLU-minion converted back to fastq
  [2023-09-11 16:06:23]	IRMA/FLU-minion saved unmatched read patterns
  [2023-09-11 16:06:23]	IRMA/FLU-minion skipping final assembly, no reference files found
  [2023-09-11 16:06:24]	IRMA/FLU-minion moving project
  [2023-09-11 16:06:24]	IRMA/FLU-minion finished!

Command error:
  cat: can't open 'GEN23-RTPCR-0829-NTC-3-1b/amended_consensus/*.fa': No such file or directory
```

### Workflow Version

3.3.4, revision: bda4dc7

### Nextflow Version

23.04.1

### Java Version

openjdk version "17.0.3-internal" 2022-04-19
OpenJDK Runtime Environment (build 17.0.3-internal+0-adhoc..src)
OpenJDK 64-Bit Server VM (build 17.0.3-internal+0-adhoc..src, mixed mode, sharing)

### Hardware

HPC Cluster

### Operating System (OS)

Distributor ID: CentOS Description: CentOS Linux release 7.9.2009 (Core) Release: 7.9.2009 Codename: Core

### Conda/Container Engine

Singularity

### Additional context

[slurm-27894603_mod.txt](https://github.com/CFIA-NCFAD/nf-flu/files/12580581/slurm-27894603_mod.txt)
[nextflow_mod.log](https://github.com/CFIA-NCFAD/nf-flu/files/12580598/nextflow_mod.log)

**Comment by peterk87** [comment]
Hi @Codes1985 Thanks for reporting this issue! The fix seems to be fairly straightforward (see PR #48)

Would you be able to provide the NTC sequences? They might make a good addition to the current CI test dataset to ensure that the pipeline can properly handle negative control sequences. I'll see if a completely different virus's sequences trigger this issue with the current version of the pipeline.

**Comment by Codes1985** [comment]
Hi @peterk87,

I have sent a secured download link to your institution email comprising example NTC read sets and a biref explanation of how they are currently behaving with nf-flu v3.3.4..

As always, thank you for your support.

**Comment by peterk87** [comment]
Hi @Codes1985  I've released [3.3.5](https://github.com/CFIA-NCFAD/nf-flu/releases/tag/3.3.5) which a fix for this issue. The negative control datasets are now used as part of the automated testing to ensure no regressions.

If you still encounter this issue, feel free to reopen this issue!
